lib/ws_servers/api/session.js

```diff
@@ -114,7 +120,14 @@ class Session {
    * @param {MetricsClient} metricsClient
    */
   setMetricsClient (metricsClient) {
-    this.services[this.mode].metricsClient = metricsClient
+    if (this.metricsClient) {
```

Comment:
also, need to add a condition in `closeServices` method to close the metrics client if it's available and open.
`if(this.metricsClient && this.metricsClient.isOpen) this.metricsClient.close()`

Reply:
thanks for catching that